src/sql/workbench/services/objectExplorer/browser/objectExplorerService.ts

```diff
@@ -276,35 +281,84 @@ export class ObjectExplorerService implements IObjectExplorerService {
 	private async handleSessionCreated(session: azdata.ObjectExplorerSession): Promise<void> {
 		let connection: ConnectionProfile | undefined = undefined;
 		let errorMessage: string | undefined = undefined;
-		if (this._sessions[session.sessionId!]) {
-			connection = this._sessions[session.sessionId!].connection;
+		const sessionId = session.sessionId;
 
-			try {
-				if (session.success && session.rootNode) {
-					let server = this.toTreeNode(session.rootNode, undefined);
-					server.connection = connection;
-					server.session = session;
-					this._activeObjectExplorerNodes[connection!.id] = server;
+
+		await new Promise<void>((resolve, reject) => {
+			const cleanup = () => {
+				if (connection) {
+					this._connectionsWaitingForSession.delete(connection.id);
 				}
-				else {
-					errorMessage = session && session.errorMessage ? session.errorMessage : errSessionCreateFailed;
-					this.logService.error(errorMessage);
+				createNewSessionListener.dispose();
+				clearTimeout(timeoutHandle);
+			}
+			const onTimeout = () => {
+				if (!this._sessions[sessionId]) {
+					this.logService.error(`Timed out waiting for session ${sessionId} to be created.
+					This has probably happened because OE service did not recieve a response for createNewSession from the provider.`);
+					reject(new Error(
+						nls.localize('objectExplorerMissingSession',
+							'Timed out waiting for session {0} to be created. This has probably happened because OE service did not recieve a response for createNewSession from the provider.', sessionId)));
```

Comment:
The user isn't going to know what the session ID means, and the reasoning why isn't something they would understand either so I think we should just simplify it.

```suggestion
							'Timed out waiting for the Object Explorer session to be created.')));
```